### ISSUE TITLE
Add ``unyt`` support to custom mixing rules

### ISSUE THREAD

**Issue by rsdefever** [MaginnGroup/mosdef_cassandra#64]
**Bug summary**
Custom mixing rules do not currently use ``unyt``. We need to reformulate
the ``custom_mixing_dict`` to use the ``unyt`` package.

**Comment by rsdefever** [comment]
@rmatsum836 your PR reminded me that custom mixing rules are clunky at the moment. We should see if we can improve them.

**Comment by rmatsum836** [comment]
Agreed.  Currently if custom mixing rules are used, the user must specify all pairwise interactions, even if the majority of them are standard geometric or LB.  I think it would be a nice feature to allow the user to only specify the custom interactions, and then infer/write out the other pairwise interactions from the underlying Parmed Structure.
